server/src/main/java/org/opensearch/action/update/TransportUpdateAction.java

```diff
@@ -276,6 +279,15 @@ protected void shardOperation(final UpdateRequest request, final ActionListener<
                 IndexRequest indexRequest = result.action();
                 // we fetch it from the index request so we don't generate the bytes twice, its already done in the index request
                 final BytesReference indexSourceBytes = indexRequest.source();
+                final Settings indexSettings = indexService.getIndexSettings().getSettings();
+                if (IndexSettings.DEFAULT_PIPELINE.exists(indexSettings) || IndexSettings.FINAL_PIPELINE.exists(indexSettings)) {
+                    deprecationLogger.deprecate(
+                        "update_operation_with_ingest_pipeline",
+                        "the index ["
+                            + indexRequest.index()
+                            + "] has a default ingest pipeline or a final ingest pipeline, but performing update operation with ingest pipeline causes unexpected result, this support will be removed in 3.0.0"
```

Comment:
I think the message is bit confusing, may be something along these lines:

```
 "the index ["+ indexRequest.index() + "] has a default ingest pipeline or a final ingest pipeline, the support of the ingest pipelines for update operation causes unexpected result and will be removed in 3.0.0"
```

My take on this is that the default / final ingest pipelines are agnostic to the fact that update request is treated as index request, so user cannot do anything about it, right? (removing these pipelines is not an option since those could be used for ingestion of new documents). 

The doubt for me here is the deprecation warning vs warning in logs (or may be both). Could user optout from this behavior now? (apply pipeline for index request, skip for update requests?)

Reply:
Thanks, I've changed the message.

> so user cannot do anything about it, right? 

Yes, users can do nothing for this.

> Could user optout from this behavior now? (apply pipeline for index request, skip for update requests?)

No, we don't have this option for users, but this behavior exists in Bulk API, but not in Update API because update request is converted to an index request.